src/VisualStudio/Core/Def/Implementation/Experimentation/KeybindingResetDetector.cs

```diff
@@ -217,37 +260,69 @@ private void ShowGoldBar()
         {
-            AssertIsForeground();
-
             // Quick exit if resharper is either uninstalled or not enabled
-            if (!_resharperExtensionEnabled)
+            if (!_resharperExtensionInstalledAndEnabled)
             {
                 return ReSharperStatus.NotInstalledOrDisabled;
             }
 
-            if (_oleCommandTarget == null)
-            {
-                _oleCommandTarget = _serviceProvider.GetService<IOleCommandTarget, SUIHostCommandDispatcher>();
-            }
+            await EnsureOleCommandTargetAsync().ConfigureAwait(false);
 
             var cmds = new OLECMD[1];
-            cmds[0].cmdID = SuspendId;
+            cmds[0].cmdID = ResumeId;
             cmds[0].cmdf = 0;
 
-            var hr = _oleCommandTarget.QueryStatus(ReSharperCommandGroup, (uint)cmds.Length, cmds, IntPtr.Zero);
-            if (ErrorHandler.Failed(hr))
+            for (var count = 0; count < 10; count++)
             {
-                // In the case of an error when attempting to get the status, pretend that ReSharper isn't enabled. We also
-                // shut down monitoring so we don't keep hitting this.
-                FatalError.ReportWithoutCrash(Marshal.GetExceptionForHR(hr));
-                Shutdown();
-                return ReSharperStatus.NotInstalledOrDisabled;
+                cancellationToken.ThrowIfCancellationRequested();
+
+                var hr = await QueryStatusOnUIThreadAsync().ConfigureAwait(false);
+                if (ErrorHandler.Failed(hr))
+                {
+                    // In the case of an error when attempting to get the status, pretend that ReSharper isn't enabled. We also
+                    // shut down monitoring so we don't keep hitting this.
+                    FatalError.ReportWithoutCrash(Marshal.GetExceptionForHR(hr));
+                    await ShutdownAsync().ConfigureAwait(false);
+
+                    return ReSharperStatus.NotInstalledOrDisabled;
+                }
+
+                // When ReSharper is suspended, the ReSharper_Resume command has the Enabled | Supported flags. 
+                if (((OLECMDF)cmds[0].cmdf).HasFlag(OLECMDF.OLECMDF_ENABLED))
+                {
+                    return ReSharperStatus.Suspended;
+                }
+
+                //otherwise sleep for a bit and check again
+                await Task.Delay(TimeSpan.FromSeconds(2), cancellationToken).ConfigureAwait(false);
+            }
+
+            // If resume button doesn't become active within a reasonable amount of time, assume ReSharper is Enabled
+            return ReSharperStatus.Enabled;
+
+            async Task<int> QueryStatusOnUIThreadAsync()
+            {
+                await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+                return _oleCommandTarget.QueryStatus(ReSharperCommandGroup, (uint)cmds.Length, cmds, IntPtr.Zero);
```

Comment:
💡  We need something like this to ensure we do not proceed with the `QueryStatus` before ReSharper finishes loading.

```csharp
var shell = await _serviceProvider.GetServiceAsync<IVsShell7, SVsShell>();
await shell.LoadPackageAsync(ReSharperPackageId);
```

/cc @AArnott for guidance on this unavoidable forced package load

Reply:
➡️ Should be possible without package load by waiting for either Suspend or Resume to no longer be *invisible*.